CDTDatastore/HTTP/CDTURLSession.m

```diff
@@ -61,7 +62,12 @@ - (instancetype)initWithCallbackThread:(NSThread *)thread
         if ([[NSURLSessionConfiguration class] respondsToSelector:@selector(backgroundSessionConfigurationWithIdentifier:)]) {
             config = [NSURLSessionConfiguration backgroundSessionConfigurationWithIdentifier:sessionId];
         } else {
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+            // since the method is only called on platforms where it's replacement is missing
+            // we supress the warning
```

Comment:
Sadly I don't think so, I had this discussion with @brynh when we first added support for replication polices, I'll do some more research to double check though